src/common/LoadingBanner.js

```diff
@@ -0,0 +1,92 @@
+/* @flow strict-local */
+
+import React, { PureComponent } from 'react';
+import { StyleSheet, View } from 'react-native';
+
+import type { Dispatch } from '../types';
+import { connect } from '../react-redux';
+import { getLoading } from '../selectors';
+import { Label, LoadingIndicator } from '.';
+import type { ThemeColors } from '../styles';
+import { ThemeContext } from '../styles';
+
+const key = 'LoadingBanner';
+
+const styles = StyleSheet.create({
+  block: {
+    flexDirection: 'row',
+    justifyContent: 'center',
+    alignItems: 'center',
+    backgroundColor: 'hsl(6, 98%, 57%)',
+  },
+  none: { display: 'none' },
+});
+
+type SelectorProps = $ReadOnly<{|
+  loading: boolean,
+|}>;
+
+type Props = $ReadOnly<{|
+  spinnerColor?: 'black' | 'white' | 'default',
+  textColor?: string,
+  backgroundColor?: string,
+
+  dispatch: Dispatch,
+  ...SelectorProps,
+|}>;
+
+/**
+ * Displays a notice that the app is connecting to the server.
+ * Rendered when session.loading is true (as of 2020/02, only
+ * during the /register request).
+ *
+ * @prop loading
+ */
+class LoadingBanner extends PureComponent<Props> {
+  static contextType = ThemeContext;
+  context: ThemeColors;
+
+  render() {
+    const {
+      loading,
+      spinnerColor = 'default',
+      textColor = this.context.color,
+      backgroundColor,
+    } = this.props;
+    if (!loading) {
+      return <View key={key} style={styles.none} />;
+    }
+    const style = {
+      width: '100%',
+      flexDirection: 'row',
+      justifyContent: 'center',
+      alignItems: 'center',
+      backgroundColor: backgroundColor ?? this.context.backgroundColor,
+    };
+    return (
+      <View key={key} style={style}>
+        <View
+          /* eslint-disable react-native/no-inline-styles */
+          style={{ flexDirection: 'row', alignItems: 'center' }}
+        >
+          <View>
+            <LoadingIndicator size={14} color={spinnerColor} />
+          </View>
```

Comment:
Ah, so it turns out that `LoadingIndicator`'s `View` has `flex: 1`. The wrapper `View` here prevents the LoadingIndicator's box from stretching out to occupy as much of the banner as it can, with "Connecting..." fitting on the right, like so:

![image](https://user-images.githubusercontent.com/22248748/75289733-c7cc6a80-57d3-11ea-9e84-13e6b0e28ab2.png)

Reply:
Got it, that makes sense. Thanks!


> Perhaps `LoadingIndicator`'s outermost `View` should not have `flex: 1`, and that way, we could remove this puzzling `View`. But we'd need to review all uses of `LoadingIndicator` to make sure it doesn't break things elsewhere.

Yeah, agreed on both counts.

This smells to me like an indication that it doesn't make sense for `LoadingIndicator` to have `flex: 1` on its outermost component. That's a property that's really about how the *parent* gets laid out, and how the given component participates in that -- so when a given component has that at its top level, it becomes effectively part of its interface, and I don't see a good reason why that'd naturally be part of the interface of `LoadingIndicator`.

Reply:
Makes sense; as mentioned in person, this will be an important fix, but outside this PR's scope.